stdlib/UUIDs/src/UUIDs.jl

```diff
@@ -157,4 +157,43 @@ function uuid5(ns::UUID, name::String)
     return UUID(v)
 end
 
+"""
+    uuid7([rng::AbstractRNG]) -> UUID
+
+Generates a version 7 (random or pseudo-random) universally unique identifier (UUID),
+as specified by RFC 9652.
+
+The default rng used by `uuid7` is not `Random.default_rng()` and every invocation of `uuid7()` without
+an argument should be expected to return a unique identifier. Importantly, the outputs of
+`uuid7` do not repeat even when `Random.seed!(seed)` is called. Currently (as of Julia 1.12),
+`uuid7` uses `Random.RandomDevice` as the default rng. However, this is an implementation
+detail that may change in the future.
+
+!!! compat "Julia 1.12"
+    `uuid7()` is available as of Julia 1.12.
+
+# Examples
+```jldoctest; filter = r"[a-z0-9]{8}-([a-z0-9]{4}-){3}[a-z0-9]{12}"
+julia> using Random
+
+julia> rng = Xoshiro(123);
+
+julia> uuid7(rng)
+UUID("019026ca-e086-772a-9638-f7b8557cd282")
+```
+"""
+function uuid7(rng::AbstractRNG=Random.RandomDevice())
+    bytes = rand(rng, UInt128)
+    # make space for the timestamp
+    bytes &= 0x0000000000000fff3fffffffffffffff
```

Comment:
is this faster than only generating the random bits that we need?

Reply:
We need a total of 12+62 random bits (see [here](https://www.rfc-editor.org/rfc/rfc9562#name-uuid-version-7)), so just picking 128 since we're generating 64 at a time anyways seems reasonable.

Also, the bottleneck here is the `ccall` required for retrieving the current `time()`, as well as the potentially-throwing path in `round`.